kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -390,17 +402,44 @@ public LogSegment getLogSegmentForVersion(Engine engine, Optional<Long> versionT
     /////////////////////////////////////////////////////////////////////////////////////////////
     // Step 7: Grab all deltas in range [$latestCompleteCheckpointVersion + 1, $versionToLoad] //
     /////////////////////////////////////////////////////////////////////////////////////////////
-    final List<FileStatus> deltasAfterCheckpoint =
+
+    final List<FileStatus> listedDeltasAfterCheckpoint =
         listedDeltaFileStatuses.stream()
             .filter(
                 fs -> {
-                  final long deltaVersion = FileNames.deltaVersion(new Path(fs.getPath()));
+                  final long deltaVersion = FileNames.deltaVersion(fs.getPath());
                   return latestCompleteCheckpointVersion + 1 <= deltaVersion
-                      && deltaVersion <= versionToLoadOpt.orElse(Long.MAX_VALUE);
+                      && deltaVersion <= versionToLoad;
                 })
             .collect(Collectors.toList());
 
-    logDebugFileStatuses("deltasAfterCheckpoint", deltasAfterCheckpoint);
+    logDebugFileStatuses("listedDeltasAfterCheckpoint", listedDeltasAfterCheckpoint);
+
+    logDebugParsedLogDatas("parsedLogDatas", parsedLogDatas);
+
+    final long suffixCommitsLowerBoundExclusive =
```

Comment:
Should we add a comment that this works because we rely on _in order backfill_? Mostly so we know if we ever relax that requirement that we'll have to come and change stuff here?

Reply:
There's a method doc on thix method already that says we only support ParsedLogData of type `RATIFIED_STAGED_COMMIT` --> I can add this extra note there, too 👍